numba/np/arrayobj.py

```diff
@@ -5257,15 +5253,26 @@ def expand_dims(context, builder, sig, args, axis):
 @intrinsic
 def np_expand_dims(typingctx, a, axis):
     layout = a.layout if a.ndim <= 1 else 'A'
-    ret = a.copy(ndim=a.ndim + 1, layout=layout)
+
+    if isinstance(axis, types.UniTuple) and axis.count:
+        ret = a.copy(ndim=a.ndim + axis.count, layout=layout)
+    elif isinstance(axis, types.Integer):
+        ret = a.copy(ndim=a.ndim + 1, layout=layout)
+    else:
+        # An empty tuple
+        ret = a.copy(ndim=a.ndim, layout=layout)
     sig = ret(a, axis)
 
     def codegen(context, builder, sig, args):
-        axis = context.cast(builder, args[1], sig.args[1], types.intp)
-        axis = _normalize_axis(context, builder, "np.expand_dims",
-                               sig.return_type.ndim, axis)
-
-        ret = expand_dims(context, builder, sig, args, axis)
+        axes_types = sig.args[1]
+        if (
```

Comment:
The formatting here seems a little off?

Reply:
Resolved in [4e4c820](https://github.com/numba/numba/pull/7604/commits/4e4c8200aae40b629579780fdcd1aa8b64096536)